slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -238,9 +238,10 @@ implicit def $name(implicit $dependencies): GR[${TableClass.elementType}] = GR{
       }
       def code = {
         val parentsString = parents.map(" with " + _).mkString("")
-        val args = model.name.schema.map(n => s"""Some("$n")""") ++ Seq("\"" + model.name.table + "\"")
+        val schema = model.name.schema.map(n => s"""Some("$n")""")
+        val quotedName = "\"" + model.name.table + "\""
         s"""
-class $name(_tableTag: Tag) extends profile.api.Table[$elementType](_tableTag, ${args.mkString(", ")})$parentsString {
+class $name(_tableTag: Tag, _tableSchema: Option[String] = $schema, _tableName: String = $quotedName) extends profile.api.Table[$elementType](_tableTag, _tableSchema, _tableName)$parentsString {
```

Comment:
two params to three params? is the constructor of `profile.api.Table[$elementType]` a polymorphic function?

Reply:
it has two constructors